README.md

```diff
@@ -26,118 +29,187 @@ GQL architecture is inspired by `React-Relay` and `Apollo-Client`.
+
+#### Using a provided schema
+
+The schema can be provided as a String (which is usually stored in a .graphql file):
 
 ```python
-from gql import Client
-from gql.transport.requests import RequestsHTTPTransport
+with open('path/to/schema.graphql') as f:
+    schema_str = f.read()
+
+client = Client(schema=schema_str)
+```
+
+OR can be created using python classes:
 
+```python
 from .someSchema import SampleSchema
+# SampleSchema is an instance of GraphQLSchema
 
-client = Client(transport=RequestsHTTPTransport(
-     url='/graphql', headers={'Authorization': 'token'}), schema=SampleSchema)
+client = Client(schema=SampleSchema)
 ```
 
-To execute against a graphQL API. (We get the schema by using introspection).
+See [tests/starwars/schema.py](tests/starwars/schema.py) for an example of such a schema.
+
+#### Using introspection
+
+In order to get the schema directly from the GraphQL Server API using the transport, you just need
+to set the `fetch_schema_from_transport` argument of Client to True
+
+### HTTP Headers
+
+If you want to add additional http headers for your connection, you can specify these in your transport:
 
 ```python
-from gql import gql, Client
-from gql.transport.requests import RequestsHTTPTransport
+transport = AIOHTTPTransport(url='YOUR_URL', headers={'Authorization': 'token'})
+```
+
+### GraphQL variables
+
+You can also provide variable values with your query:
 
-sample_transport=RequestsHTTPTransport(
-    url='https://countries.trevorblades.com/',
-    verify=False,
-    retries=3,
+```python
+query = gql(
+    """
+    query getContinentName ($code: ID!) {
+      continent (code: $code) {
+        name
+      }
+    }
+"""
 )
 
+params = {"code": "EU"}
+
+# Get name of continent with code "EU"
+result = client.execute(query, variable_values=params)
+print(result)
+
+params = {"code": "AF"}
+
+# Get name of continent with code "AF"
+result = client.execute(query, variable_values=params)
+print(result)
+```
+
+### GraphQL subscriptions
+
+Using the websockets transport, it is possible to execute GraphQL subscriptions:
+
+```python
+from gql import gql, Client, WebsocketsTransport
+
+transport = WebsocketsTransport(url='wss://your_server/graphql')
+
 client = Client(
-    transport=sample_transport,
+    transport=transport,
     fetch_schema_from_transport=True,
 )
 
 query = gql('''
-    query getContinents {
-      continents {
-        code
-        name
-      }
+    subscription yourSubscription {
+        ...
     }
 ''')
 
-client.execute(query)
+for result in client.subscribe(query):
+    print (result)
 ```
 
-If you have a local schema stored as a `schema.graphql` file, you can do:
+> **Note**: the websockets transport can also execute queries or mutations, it is not restricted to subscriptions
+
+### Execute on a local schema
+
+It is also possible to execute queries against a local schema (so without a transport).
```

Comment:
Since I mentioned how the basic usage would be without the transport, maybe we should move this section after basic usage?

Perhaps mention that the basic usage uses a local transport and then show the example with transports?